drivers/flash/flash_stm32_ospi.c

```diff
@@ -276,7 +275,9 @@ static int ospi_read_sfdp(const struct device *dev, off_t addr, uint8_t *data,
 	}
 	cmd.Address = addr;
 	cmd.NbData = size;
-
+	cmd.AddressSize = ((dev_cfg->data_mode == OSPI_SPI_MODE)
+				?  HAL_OSPI_ADDRESS_24_BITS
+				:  HAL_OSPI_ADDRESS_32_BITS);
```

Comment:
Why 32 bits for everything except SPI?
What about dual and quad modes? Many ICs there expect 24 bits.
Possible solution:
```suggestion
	cmd.AddressSize = ((dev_cfg->data_mode == OSPI_OPI_MODE)
				?  HAL_OSPI_ADDRESS_34_BITS
				:  HAL_OSPI_ADDRESS_24_BITS);
```

Reply:
ok for changing order to 
```
	cmd.AddressSize = ((dev_cfg->data_mode == OSPI_OPI_MODE)
				?  HAL_OSPI_ADDRESS_32_BITS
				:  HAL_OSPI_ADDRESS_24_BITS);
```